### ISSUE TITLE
Null reference exception in MainToolbarController

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#5940]
Build Information
VS Mac 7.8.0.444
Git revision: 42be91afe3ba525f327837045ca91fd2c834cd7b
Build date: 2018-09-12 06:07:51+00
Build branch: master
Xamarin extensions: 14685af820b66440e7a955d9a31c4431e592ba6b

Seeing the following null reference logged several times in the IDE log.

```
 An unhandled exception has occurred. Terminating Visual Studio? False
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.Components.MainToolbar.MainToolbarController.<HandleSearchEntryChanged>b__48_1 (System.Object <p0>, System.EventArgs <p1>) [0x00000] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/MainToolbarController.cs:699 
  at MonoDevelop.Components.MainToolbar.SearchPopupWidget.set_SelectedItem (MonoDevelop.Components.MainToolbar.SearchPopupWidget+ItemIdentifier value) [0x00007] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/SearchPopupWindow.cs:885 
  at MonoDevelop.Components.MainToolbar.SearchPopupWidget.Dispose (System.Boolean disposing) [0x00053] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/SearchPopupWindow.cs:220 
  at System.ComponentModel.Component.Dispose () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System/compmod/system/componentmodel/Component.cs:119 
  at (wrapper remoting-invoke-with-check) System.ComponentModel.Component.Dispose()
  at Xwt.Widget.Dispose (System.Boolean disposing) [0x0003f] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/external/xwt/Xwt/Xwt/Widget.cs:322 
  at System.ComponentModel.Component.Dispose () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System/compmod/system/componentmodel/Component.cs:119 
  at (wrapper remoting-invoke-with-check) System.ComponentModel.Component.Dispose()
  at Xwt.Window.Dispose (System.Boolean disposing) [0x0000b] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/external/xwt/Xwt/Xwt/Window.cs:147 
  at System.ComponentModel.Component.Dispose () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System/compmod/system/componentmodel/Component.cs:119 
  at (wrapper remoting-invoke-with-check) System.ComponentModel.Component.Dispose()
  at MonoDevelop.Components.MainToolbar.SearchPopupWidget+<OpenFile>d__23.MoveNext () [0x00283] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/SearchPopupWindow.cs:252 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1018 
  at MonoDevelop.Ide.DispatchService+GtkSynchronizationContext+TimeoutProxy.HandlerInternal (System.IntPtr data) [0x00014] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/DispatchService.cs:77 

```

Similar callstack reported by Alan - https://gist.github.com/alanmcgovern/e11855a8dc659d49b40a4187bb424214

> VS bug [#682796](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/682796)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-21] Comment by **xamarinc**



Fixed in version 7.8.0.1109 (master)



Author: Vsevolod Kukol  
 Commit: 11d413a1ac05eb4d497a03f886790d62306deb5c (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-24] Comment by **xamarinc**



Fixed in version 8.0.0.604 (release-8.0)



Author: Vsevolod Kukol  
 Commit: 11d413a1ac05eb4d497a03f886790d62306deb5c (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-25] Comment by **xamarinc**



Fixed in version 7.8.0.1193 (release-7.8)



Author: Vsevolod Kukol  
 Commit: 11d413a1ac05eb4d497a03f886790d62306deb5c (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-28] Comment by **xamarinc**



Fixed in version 7.7.1.6 (release-7.7)



Pull Request #6542 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 88dcd076c40f0591ac359682459c751cff270807 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-30] Comment by **xamarinc**



Fixed in version 8.0.0.1738 (release-8.0)



Author: Vsevolod Kukol  
 Commit: 11d413a1ac05eb4d497a03f886790d62306deb5c (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-37] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Author: Vsevolod Kukol  
 Commit: 11d413a1ac05eb4d497a03f886790d62306deb5c (xamarin/monodevelop)
